mobile/lib/src/infrastructure/bluetooth_repository.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:async';
+
+import 'package:flutter_blue/flutter_blue.dart';
+import 'package:polydodo/src/domain/domain.dart';
+
+class BluetoothRepository implements IBluetoothRepository {
+  static const String BLE_SERVICE = "fe84";
+  static const String BLE_RECEIVE = "2d30c082";
+  static const String BLE_SEND = "2d30c083";
+  static const startStreamChar = 'b';
+  static const stopStreamChar = 's';
+
+  BluetoothDevice selectedDevice;
+  BluetoothCharacteristic sendCharacteristic;
+  BluetoothCharacteristic receiveCharacteristic;
+
+  static List<BluetoothDevice> bluetoothPersistency = [];
+  final streamController = StreamController<List<BluetoothDevice>>();
+
+  void initializeBluetooth() {
+    FlutterBlue flutterBlue = FlutterBlue.instance;
+    flutterBlue.connectedDevices
+        .asStream()
+        .listen((List<BluetoothDevice> devices) {
+      for (BluetoothDevice device in devices) {
+        addDevice(device);
+      }
+    });
+    flutterBlue.scanResults.listen((List<ScanResult> results) {
+      for (ScanResult result in results) {
+        addDevice(result.device);
+      }
+    });
+    flutterBlue.startScan();
+  }
+
+  Future<void> addDevice(BluetoothDevice device) async {
+    await Future.delayed(Duration(milliseconds: 400));
```

Comment:
pourquoi est ce que l'on doit attendre 400 ms?

Reply:
Bad case of copy pasting William's example. Removed it and works fine without it.